lib/fs.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const mkdir = (dir, mode, cb) => {
+  fs.access(dir, fs.constants.F_OK, err => {
+    if (err && err.code === 'ENOENT') {
+      mkdir(path.dirname(dir), mode, err => {
+        if (err) cb(err);
+        else fs.mkdir(dir, cb);
+      });
+    } else {
+      cb(err);
+    }
+  });
+};
+
+const addMode = (mode, cb) => {
+  if (typeof mode === 'function') {
+    cb = mode;
+    mode = '0777';
+  }
+  return [mode, cb];
+};
+
+let mkdirp;
+const version = process.versions.node.split('.').map(el => parseInt(el));
+if (version[0] < 10 || (version[0] === 10 && version[1] <= 11)) {
+  mkdirp = (dir, mode, cb) => {
+    [mode, cb] = addMode(mode, cb);
```

Comment:
```suggestion
  if (typeof mode === 'function') {
    cb = mode;
    mode = '0777';
  }
```
IMO, it will be better to inline this.